EmployeeDepartment/src/main/java/com/example/EmployeeDepartment/ServiceImpl/EmployeeServiceImpl.java

```diff
@@ -0,0 +1,80 @@
+package com.example.EmployeeDepartment.ServiceImpl;
+
+import com.example.EmployeeDepartment.DAO.DepartmentRepository;
+import com.example.EmployeeDepartment.DAO.EmployeeRepository;
+import com.example.EmployeeDepartment.entity.Department;
+import com.example.EmployeeDepartment.entity.Employee;
+import com.example.EmployeeDepartment.services.EmployeeService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class EmployeeServiceImpl implements EmployeeService {
+    @Autowired
+    private EmployeeRepository employeeRepository;
+    @Autowired
+    private DepartmentRepository departmentRepository;
+
+    @Override
+    public List<Employee> getEmployees() {
+        return employeeRepository.findAll();
+    }
+
+    @Override
+    public Employee addEmployee(Employee employee, int id) {
+        Department department = departmentRepository.findById(id).orElse(null);
+        if (null != department) {
+            employee.setDepartment(department);
+        }
+        //save employee
+        Employee emp = employeeRepository.save(employee);
+        //get list of employees for that department
+        List<Employee> employeeList = emp.getDepartment().getEmployees();
```

Comment:
The object updation should not be done like this from the code. Should be updated from the DB itself